Cookbook/Proposals/Comments.md

```diff
@@ -0,0 +1,223 @@
+    
+I think that adding the above section to our style guide would make it clear what's our approach to comments.
+
+### What is a good comment?
+
+To make our StyleGuide more verbose we can also add a list of what kind of comments we think are "good".
+
+**Good comments**:
+
+- **TODO comments**
+    This already has a good definition of our approach in the StyleGuide and I think we are fine with that approach.
+- **Doc headers for public API**
+    Good documentation helps consumers of the SDK to understand what the code does. However, I don't know if it's worth to add a comment to every public function/field, even if a comment is just a repetition. The following comment doesn't add any value (unless it's needed by a doc generation tool if we use any):
+    ```swift
+     /// Supported biometric types
+     public var supportedBiometry: BiometryType
+    ```
+- **Explanation of Intent/Clarification**
+    It's a kind of comment when a developer tries to explain what he's trying to achieve when a code is not verbose enough. Example: Math code, sometimes parsing.
+- **Warning of Consequences**
+    If we had a test which takes an hour to run, it would be nice to have a comment warning about it.
+- **Amplification**
+    Sometimes some piece of code seems to be less important than it really is. In such cases, it's worth to mark it.
+- **Reference to external sources**
+    For example, ISO 639 for country codes.
+
+The above list is what defines a good comment. **However**, we still have to remember that **every** comment **is our failure** to name our code better. If someone gives you a good hint how to remove a comment without sacrificing readability **we should go for it**.
+
+### How can we replace a comment?
+In case of a **what** comments usually it is very easy to replace a comment. Usually, it's all about taking what comment says and create a variable or function with such a name.
+
+
+### How do we comment?
+Even with the proposed approach, we will be forced to comment our code at some point. I would like to propose a way of doing it. The biggest difficulty for me when I read a comment (especially a long one, explaining "why") is to understand to what scope the comment refers to. I think that we should always try to export commented code into a separate method/class and put the comment as a header to this method.
+
+The above solution has 2 benefits:
+
+1. It's visible to what part of a code comment refers.
+2. It improves readability for a developer who doesn't need to know why a particular piece of code is written. Sometimes, during debugging for example, what you need to know is **what** happens, not **why**.
+
+To illustrate point `2.`. If there is an UIKit bug, with dismissing a modal screen which is ugly but works just fine, all we care about is the fact that the code has been dismissed. During a debuting we now we can go just further without a need to understand how `dismissing` is handled.
```

Comment:
Didn't understand the `debuting` bit; can you please clarify?

Reply:
@NSMyself it's likely a typo: https://github.com/Babylonpartners/ios-playbook/pull/132#discussion_r284149699

Reply:
That would make sense indeed, thanks Olivier

Reply:
yes, typo ;)